includes/amp-helper-functions.php

```diff
@@ -653,12 +653,14 @@ function amp_get_current_url() {
 	if ( isset( $parsed_url['port'] ) ) {
 		$current_url .= ':' . $parsed_url['port'];
 	}
-	$current_url .= '/';
+	
+	$current_url .= '/' . trim($parsed_url['path'], '/') . '/' ;
 
 	if ( isset( $_SERVER['REQUEST_URI'] ) ) {
 		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
-		$current_url .= ltrim( wp_unslash( $_SERVER['REQUEST_URI'] ), '/' );
+		$current_url .= ltrim( wp_unslash( str_replace($parsed_url['path'], '', $_SERVER['REQUEST_URI']) ), '/' );
```

Comment:
Also, wouldn't it be a good hardening to ensure that it only removes `$parsed_url['path']` from the string only when it appears at the very beginning? Let's say my install is located at `/wordpress/` but the `REQUEST_URI` is `/wordpress/category/wordpress/` then this would result in `/category/` being incorrectly appended to `$current_url` as opposed to `/category/wordpress/`.